nodes/signal_shutdown.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python
+# Simple script for shutting down the raspberry Pi at the press of a button.
+# by Inderpreet Singh
+# Adapted by Jake Rye and Bob Davis
+
+import RPi.GPIO as GPIO
+import time
+import os
+import logging
+import rospy
+from std_msgs.msg import Float64
+
+# This is needed to account for any erroraneous drops in the shutdown signal
+VERIFICATION_TIME_SEC = 3          # Time to wait to verify shutdown signal
+SHUTDOWN_SIGNAL_THRESHOLD = 0.98   # Percent of shutdown signals needed to
+                                   # signal a shutdown (ranges from 0 to 1)
+
+
+def setup_gpio_pins():
+    # Use the Broadcom SOC Pin numbers
+    # Setup the Pin with Internal pullups enabled and PIN in reading mode.
+    rospy.logdebug('Setting GPIO27 / PIN13 to INPUT PULLUP')
+    GPIO.setmode(GPIO.BOARD)
+    # PIN13 / BCM 27 YELLOW / SW-COM - Signals Shutdown
+    GPIO.setup(13, GPIO.IN, pull_up_down = GPIO.PUD_UP)
+
+    # Turn delay relay on
+    # Sends signal to the relay to stay on until this drops low.
+    rospy.logdebug('Setting GPIO22 / PIN15 to OUTPUT HIGH')
+    GPIO.setup(15, GPIO.OUT) # PIN15 / BCM 22 / WHITE / CH1
+    GPIO.output(15, GPIO.HIGH)
+
+
+def check_for_shutdown(pub13):
+    # Monitor pin for stable signal to safely shutdown
+    while not rospy.is_shutdown():
+        pub13.publish(GPIO.input(13))
```

Comment:
Good point. This was mainly put in for debugging and left it thinking it could be useful to have in the future to help debugging. I'll move it over to the logdebug.